### PR TITLE
IGNITE-19874 GridDhtTxLocal initial cleanup

### PR DESCRIPTION
**Pull request by anton-vinogradov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxLocal.java [hunk 1]
```diff
@@ -244,13 +244,6 @@ public void nearFutureId(IgniteUuid nearFutId) {
         return cacheCtx.isDht() && isNearEnabled(cacheCtx) && !cctx.localNodeId().equals(nearNodeId());
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxLocal.java:244]: What about adding generics for variables of `GridCacheSharedContext`, `GridCacheContext`, `GridDhtTxFinishFuture`, `IgniteInternalFuture` classes to remove warnings? At least we can add `<?>` and `<?, ?>` without any problems here.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxLocal.java [hunk 2]
```diff
@@ -94,7 +94,7 @@ public class GridDhtTxLocal extends GridDhtTxLocalAdapter implements GridCacheMa
     private volatile GridDhtTxPrepareFuture prepFut;
 
     /** Transaction label. */
-    private @Nullable String lb;
+    private final @Nullable String lb;
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxLocal.java:97]: Looks like `lb`, `prepFut` and `PREP_FUT_UPD` can be moved to upper class.

Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxLocal.java:97]: Sure, but current goal is just to apply IDEA's tips.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxLocal.java (hunk 1 above)
Comment (alex-plekhanov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxLocal.java:244]: > current goal is just to apply IDEA's tips
One of IDEA's tips is to add generics. There is nothing to refactor, just add <?> to variables declarations in places, where IDEA complains.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxLocal.java (hunk 2 above)
Comment (alex-plekhanov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxLocal.java:97]: The goal looks strange. It's a small refactoring anyway, why both of these refactorings can't be done together?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxLocal.java (hunk 1 above)
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxLocal.java:244]: Correct, but the goal is to move step by step. 
Initial goal is to finalize fields and remove everything unused.
This should help to keep patches as simple as posslible.

The main review question - is the patch ok, not - is it complete. Can we aggree on that?

**Pull request merged**